src/scope.ts

```diff
@@ -521,6 +531,7 @@ export class Scope {
         scope.localIndex = this.localIndex;
         scope.mangledName = this.mangledName;
         scope.modifiers = this.modifiers;
+        if (this.genericOwner) scope.setGenericOwner(this.genericOwner);
```

Comment:
`copy` and `clone`, which is used for generic specialization?
According to the function name, I would expect the result scope is the same as the original one (e.g. the genericOwner field should be the same), but this line breaks this assumption.
I would suggest to use `specialize` as the method name used to process generic specialization.

Reply:
done.
The name of method was changed to **specialize**